File: Python/Iniciante/1048.py
# Problema 1048 - Beecrowd - Iniciante - Nível 2

# Recebe o valor na entrada padrão
salario = float(input())

# Inicializa as variáveis que serão usadas para imprimir os valores
percentual = 0
novoSalario = 0
reajuste = 0

# Verifica em qual intervalo o salário está e
# faz as devidas atribuições às variáveis
if salario <= 400:
    percentual = 0.15
elif salario <= 800:
    percentual = 0.12
elif salario <= 1200:
    percentual = 0.10
elif salario <= 2000:
    percentual = 0.07
else:
    percentual = 0.04

novoSalario = salario + salario*percentual
reajuste = salario*percentual

# Imprime os resultados obtidos com base no salário informado
print(f"Novo salario: {novoSalario:.2f}")
print(f"Reajuste ganho: {reajuste:.2f}")
print(f"Em percentual: {(percentual*100):.0f} %")

File: Rust/Iniciante/1048.rs
use std::io;
use std::error::Error;

fn main() -> Result<(), Box<dyn Error>> {
	let stdin = io::stdin();
	let mut buffer = String::new();

	stdin.read_line(&mut buffer)?;

	let salario: f64 = buffer.trim().parse()?;

	let novo_salario: f64;
	let reajuste: f64;
	let percentual: f64;

	if salario > 2000.0 {
		percentual = 0.04;
	} else if salario > 1200.0 {
		percentual = 0.07;
	} else if salario > 800.0 {
		percentual = 0.10;
	} else if salario > 400.0 {
		percentual = 0.12;
	} else {
		percentual = 0.15;
	}

	reajuste = percentual * salario;
	novo_salario = salario + reajuste;

	println!("Novo salario: {:.2}", novo_salario);
	println!("Reajuste ganho: {:.2}", reajuste);
	println!("Em percentual: {} %", (percentual * 100.0) as i64);

	Ok(())
}